Consistent sticky sidebar behaviour

jackbentley opened the pull request:
Update the behaviour of sticky sidebar to be consistent regardless of main content height.

Additionally fixes issues with "popping" of the sidebar when using the new `tab_header` block. 
And fixes spacing issues at the bottom of the settings panel when in "sticky" mode.

Once merged, we can remove our custom code from XFP which currently allows this behaviour.

**Before:**
<img src="https://user-images.githubusercontent.com/9383439/93773717-4e391580-fc18-11ea-848d-2696b23cf155.gif" alt="scroll-before-empty" width="50%"><img src="https://user-images.githubusercontent.com/9383439/93773736-55602380-fc18-11ea-9aa2-c6b2329f0047.gif" alt="scroll-before-full" width="50%">

**After:**
<img src="https://user-images.githubusercontent.com/9383439/93773763-5db85e80-fc18-11ea-8ae7-294a126d89da.gif" alt="scroll-after-empty" width="50%"><img src="https://user-images.githubusercontent.com/9383439/93773776-63ae3f80-fc18-11ea-954d-f7f81d98b0c6.gif" alt="scroll-after-full" width="50%">

Stanton commented on stylesheets/_component.calendar.scss:
```diff
@@ -5,7 +5,6 @@
 }
 
 .ui-datepicker {
-    padding-bottom: 1em;
```
Is this relevant to this change?

jackbentley replied:
Yes and no. It causes additional padding at the bottom of every page (like the footer) which also causes additional scroll on the right-hand content as, unlike the footer, it isn't counteracted in sticky mode.

It looks like this CSS rule is overridden when the date picker shows by in-line styles. So it's essentially useless.

<img width="685" alt="Screenshot 2020-09-22 at 17 37 48" src="https://user-images.githubusercontent.com/9383439/93911574-7b5df480-fcfa-11ea-88aa-e3d6daed73e4.png">

Stanton commented on stylesheets/_layout.settings.scss:
```diff
@@ -584,19 +585,26 @@
+    .tab__inner.is-sticky {
+        min-height: 100vh;
+        padding-bottom: 0;
+        margin-bottom: -37px;
```
Magic number needs explaining in a comment, presume it's to cater for footer?

jackbentley replied:
It is indeed to cater for the footer height. Do we have a variable for this?

Stanton replied:
Have added (and wired in) a $footer-height token which does the same calculation of line height + padding + border

Stanton reviewed (changes requested):
1. Updated js methods need test cases editing/adding (see codecov failure)
2. There needs to be some visual indication that the sidebar does not contain any items underneath the footer, some additional spacing is required.

![sidebar](https://user-images.githubusercontent.com/18653/93780974-bc81d600-fc20-11ea-90ff-321c2e430e50.gif)

jackbentley commented:
> 1. Updated js methods need test cases editing/adding (see codecov failure)
> 2. There needs to be some visual indication that the sidebar does not contain any items underneath the footer, some additional spacing is required.

@Stanton Thanks for taking a look

1) Added a new test case
2) Can't replicate that. For me, the scrollbar is correct. This is what I meant by `And fixes spacing issues at the bottom of the settings panel when in "sticky" mode.`

<img width="1262" alt="Screenshot 2020-09-22 at 17 52 34" src="https://user-images.githubusercontent.com/9383439/93913213-a8aba200-fcfc-11ea-8665-7869bbc8f414.png">
